string transform_to
float32 surface_threshold
---
float32 a
float32 b
float32 c
float32 d